### ISSUE TITLE
[NOT AN ISSUE] Porting loghi-htr to pytorch

### ISSUE THREAD

**Issue by johnlockejrr** [rvankoert/loghi-htr#31]
Just a heads up for a project of mine. Trying to port Loghi-HTR to pytorch. I'll let you know about the success/insuccess ;)

<img width="2555" height="453" alt="Image" src="https://github.com/user-attachments/assets/b0166d49-b4f9-4e8c-ba13-f035c1970792" />

Unfortunatelly `CTCWordBeamSearch` is for `tensorflow` so I switched to `pyctcdecode + optional KenLM`
Unfortunatelly also, I will have to train the models from scratch because the original ones were trained on tensorflow and converting them to pytorch via onnx I don't think is a good idea.

**Comment by johnlockejrr** [comment]
Working with pytorch. Also coverted to lighter/faster, onnx and OpenVino.
Trained a model from scratch with a small dataset of `train/val/test` `30991/6037/3219`
The accuracy is greater than the groundtruth, impressive!

```Epoch 26/99  train_loss_epoch: 0.060  val_CER_metric: 0.020  val_WER_metric: 0.081```

<img width="2845" height="1537" alt="Image" src="https://github.com/user-attachments/assets/f84a0c9f-6d92-4fca-b968-77980ab32f84" />
